tests/integration_tests/test_hyperopt.py

```diff
@@ -317,5 +387,53 @@ def test_hyperopt_run_hyperopt(csv_filename, search_space, tmpdir, ray_cluster):
     # check for existence of the hyperopt statistics file
     assert os.path.isfile(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json"))
 
-    if os.path.isfile(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json")):
-        os.remove(os.path.join(tmpdir, "test_hyperopt", "hyperopt_statistics.json"))
+
+@pytest.mark.distributed
+def test_hyperopt_run_shared_params_trial_table(csv_filename, tmpdir):
+    config, rel_path, cell_types_search_space, vocab_size_search_space = _setup_ludwig_config_with_shared_params(
+        csv_filename
+    )
+
+    hyperopt_results = hyperopt(config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")
+    hyperopt_results_df = hyperopt_results.experiment_analysis.results_df
+
+    # Check that the trials did sample from defaults in the search space
+    for _, trial_row in hyperopt_results_df.iterrows():
+        vocab_size = _get_trial_parameter_value("defaults.output_features.category.vocab_size", trial_row)
+
+        cell_type = _get_trial_parameter_value("defaults.input_features.text.cell_type", trial_row)
+        if cell_type is not None:
```

Comment:
The cell type should always be non-None in this test, right?

Reply:
Yup, agreed